### PR TITLE
Add average response time endpoint

### PR DESCRIPTION
**Pull request by shaneboulden**
Adds an endpoint that returns an array of data with the average response times per 100 requests (and associated times). This can then be used by `peakweb` to display a graph of average response times back to the user.

### CODE REVIEW

**Comment by shaneboulden** [comment]
Fixed the average response time calculation logic, additional context is in the Trello ticket: https://trello.com/c/wn9ZSsIR

File: peakorc.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import falcon
 from peakmodels import *
 from playhouse.shortcuts import model_to_dict
+import numpy as np
```
Comment (ghost) [inline peakorc.py]: is shortening to np really needed? two letter names make me twitch

Comment (shaneboulden) [reply peakorc.py]: But I can reach them with one hand! You're absolutely right, I don't think it makes the code any clearer - I'll fix and push the changes.

**Comment by adamgoossens** [comment]
Some thoughts on this - rather than have an avg-time endpoint, perhaps we have a "metrics" ("results"?) endpoint beneath the suite resource, e.g.: /suite/1a6ab35b-3648-48dc-aa54-550f06d7a0be/metrics

Metrics (or whatever we call it) then returns all of the key data as a JSON dictionary - average time, max, min, etc. Since "avg-time" doesn't make sense outside of the context of a test suite it's probably not a good candidate for a resource in the API. Other time metrics like max, min, etc, would fall in the same boat.

For example, a call to metrics might produce:
`{ "suite": "1a6ab35b-3648-48dc-aa54-550f06d7a0be", "avg": 0.671, "max": 1.42, "min": 0.44 }`

Thoughts?

**Comment by shaneboulden** [comment]
I think this sounds great, and should also extend to the other existing API endpoints. Eg, '/test_suite/new' doesn't make much sense, it would be far better written as '/suite/new'.

For this specific endpoint - returning a JSON array of tuples of the form `(averaged_duration_across_all_tests, averaged_response_time)`, do you think it best to include as an individual endpoint under 'suites', or as a sub-resource under the proposed metrics endpoint ('/suites/metrics/response_times')?

**Comment by shaneboulden** [comment]
I've updated the API endpoints to use a more REST-like structure. API endpoints are currently either 'suites' or 'tests'. The average response times end-point introduced by this PR is referenced as '/suites/<suite_uuid>/metrics/avg_response_times', and still returns the same array of tuples.

Note that the reorganisation caused breaking changes, which are addressed in these `peakweb` and `peaktest` PRS:
- https://github.com/peak-oss/peakweb/pull/8
- https://github.com/peak-oss/peaktest/pull/2

File: peakorc.py [hunk 2]
```diff
@@ -172,11 +172,14 @@ def on_get(self, req, resp, suite_uuid):
 peak_test = PeakTestResource()
 peak_suite_tests = PeakSuitesTestsDetailResource()
 
-api.add_route('/test_suite/new', peak_suite_new)
+# suites
 api.add_route('/suites/', peak_suites)
+api.add_route('/suites/new', peak_suite_new)
```
Comment (adamgoossens) [inline peakorc.py]: I think we can omit this particular route. Instead, a POST request to /suites/ would create a new test suite resource and return the resources as the response to the user.

Comment (shaneboulden) [reply peakorc.py]: Completely agree, it's definitely more REST-like, and the other route was redundant. I've just updated this branch and also https://github.com/peak-oss/peakweb/pull/8, as the old route was referenced in `peakweb`. Thanks!

**Comment by adamgoossens** [comment]
Awesome! +1 from me 👍

**Pull request merged**